### PR TITLE
[DataGrid] Add 'onColumnHeaderContextMenu' event

### PR DESCRIPTION
**Pull request by vfbiby**
- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Comment by vfbiby** [comment]
hi, all ci test pass, please give a review and let it merge!

File: packages/x-data-grid/src/models/events/gridEventLookup.ts [hunk 1]
```diff
@@ -88,6 +88,13 @@ export interface GridColumnHeaderEventLookup {
     params: GridColumnHeaderParams;
     event: React.MouseEvent<HTMLElement>;
   };
+  /**
+   * Fired when a column header is double-clicked.
```
Comment (cherniavskii) [inline packages/x-data-grid/src/models/events/gridEventLookup.ts:92]:
```suggestion
   * Fired when the user attempts to open a context menu in the column header.
```

File: packages/x-data-grid/src/models/props/DataGridProps.ts [hunk 2]
```diff
@@ -542,6 +542,13 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
    * @param {GridCallbackDetails} details Additional details for this callback.
    */
   onColumnHeaderClick?: GridEventListener<'columnHeaderClick'>;
+  /**
+   * Callback fired when a double click event comes from a column header element.
```
Comment (cherniavskii) [inline packages/x-data-grid/src/models/props/DataGridProps.ts:546]:
```suggestion
   * Callback fired when a contextmenu event comes from a column header element.
```

File: packages/x-data-grid/src/models/props/DataGridProps.ts [hunk 3]
```diff
@@ -542,6 +542,13 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
...
+   * @param {GridColumnHeaderParams} params With all properties from [[GridColumnHeaderParams]].
+   * @param {MuiEvent<React.MouseEvent>} event The event object.
+   * @param {GridCallbackDetails} details Additional details for this callback.
```
Comment (cherniavskii) [inline packages/x-data-grid/src/models/props/DataGridProps.ts:549]:
```suggestion
```

**Review by cherniavskii — COMMENTED** [review]
Thank you!

**Comment by vfbiby** [comment]
@cherniavskii hi, every thing looks ok

**Review by cherniavskii — APPROVED** [review]

**Pull request merged**